nwb_conversion_tools/json_schema_utils.py

```diff
@@ -67,10 +68,24 @@ def get_schema_from_method_signature(class_method, exclude=None):
     for param in inspect.signature(class_method).parameters.values():
         if param.name not in exclude + ['self']:
             if param.annotation:
-                param_type = annotation_json_type_map[str(param.annotation).split("'")[1]]
+                anno = str(param.annotation)
+                if "typing.Union" in anno:
+                    types = re.search("typing.Union\[(.*)\]", anno).group(1).split(",")
+                    intersect_valid_keys = list(set(annotation_json_type_map.keys()).intersection(types))
```

Comment:
It would be better to use `isinstance(param.annotation, typing._GenericAlias)` and `param.annotation.__args__` rather than parsing the string

Reply:
Agree 100% on using `param.annotation.__args__` (as long as its not an `inspect._empty`) instead of the regex.

`typing._GenericAlias` might be a tad too general in terms of intended support since that includes other things we would have to add, like `List`; and would also require contingency for nesting stuff like `List[Set[Union[type, str]]] ` (which is just random example code I found related to use of  `_GenericAlias`, not code we use of course).

Another option would be the `get_origin()` function from `typing_inspect`; the syntax would be
`
typing_inspect.get_origin(param.annotation) == typing.Union
`
however, `typing_inspect` does not come in the python base and has to be `pip` installed, so maybe you would be against using that due to the extra `requirements`?